common/logisticspipes/network/packets/SetCurrentPagePacket.java

```diff
@@ -0,0 +1,76 @@
+package logisticspipes.network.packets;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumHand;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import logisticspipes.LogisticsPipes;
+import logisticspipes.items.ItemGuideBook;
+import logisticspipes.network.abstractpackets.ModernPacket;
+import logisticspipes.utils.StaticResolve;
+import network.rs485.logisticspipes.util.LPDataInput;
+import network.rs485.logisticspipes.util.LPDataOutput;
+
+/**
+ * TODO make this right
+ */
+
+@StaticResolve
+public class SetCurrentPagePacket extends ModernPacket {
+
+	@Getter
+	@Setter
+	private float sliderProgress;
+
+	@Getter
+	@Setter
+	private int page;
+
+	@Getter
+	@Setter
+	private EnumHand hand;
+
+	public SetCurrentPagePacket(int id) {
+		super(id);
+	}
+
+	@Override
+	public void processPacket(EntityPlayer player) {
+		ItemStack book;
+		if (hand == EnumHand.MAIN_HAND) {
+			if (!(player.getHeldItemMainhand().getItem().getClass() == ItemGuideBook.class)) return;
```

Comment:
No. Check against the item instance that you registered, not the class, or anything else. You actually have to save it somewhere when registering, though, which you currently aren't doing.

Reply:
`if(book.getItem() != LPItems.itemGuideBook) return;` Would this be the right way?

Reply:
@Bernasss12 yes

Reply:
Great!